cvs diff (in directory J:\wxcvs\wxWindows\)
cvs server: Diffing include/wx
Index: include/wx/defs.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/defs.h,v
retrieving revision 1.247
diff -r1.247 defs.h
1219a1220,1236
>  * wxCheckBox flags.
>  * wxCH_* because of wxCB_* is used by wxComboBox
>  * Determine, whether to use a 3State or 2State
>  * Checkbox. 3State enables to differentiate 
>  * between 'Set', 'Blank' or 'Undefined'
>  * for now, wxMSW-only
>  */
> #define wxCH_2STATE           0x0000
> #define wxCH_3STATE           0x1000
> enum wxCheckBoxState
> { 
>     wxCHK_UNDEF = -1, 
>     wxCHK_UNCHECKED, 
>     wxCHK_CHECKED 
> };
> 
> /*
cvs server: Diffing include/wx/msw
Index: include/wx/msw/checkbox.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/msw/checkbox.h,v
retrieving revision 1.10
diff -r1.10 checkbox.h
48a49,53
>     // 3State Checkbox Methods
>     virtual void SetValue(wxCheckBoxState value);
>     virtual wxCheckBoxState Get3StateValue() const;
>     virtual bool Has3State() const;
> 
54a60
>     long m_checkboxStyle;
cvs server: Diffing src/msw
Index: src/msw/checkbox.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/msw/checkbox.cpp,v
retrieving revision 1.30
diff -r1.30 checkbox.cpp
103c103,109
<     long msStyle = BS_AUTOCHECKBOX | WS_TABSTOP | WS_CHILD | WS_VISIBLE;
---
>     long msStyle;
>     
>     if ( style & wxCH_3STATE )
>         msStyle = BS_AUTO3STATE | WS_TABSTOP | WS_CHILD | WS_VISIBLE;
>     else
>         msStyle = BS_AUTOCHECKBOX | WS_TABSTOP | WS_CHILD | WS_VISIBLE;
> 
200a207,238
> }
> 
> // 3State Stuff starts here
> void wxCheckBox::SetValue(wxCheckBoxState value)
> {
>     switch (value)
>     {
>     case wxCHK_CHECKED:     SetValue(TRUE);     break;
>     case wxCHK_UNCHECKED:   SetValue(FALSE);    break;
>     case wxCHK_UNDEF:
>         {
>             wxASSERT((m_checkboxStyle & BS_AUTO3STATE) == BS_AUTO3STATE);
>             SendMessage(GetHwnd(), BM_SETCHECK, BST_INDETERMINATE, 0);
>         } break;
>     }
> }
> 
> wxCheckBoxState wxCheckBox::Get3StateValue() const
> {
> #ifdef __WIN32__
>     long lState = SendMessage(GetHwnd(), BM_GETCHECK, 0, 0);
>     if (lState == BST_CHECKED)	return wxCHK_CHECKED;
>     if (lState == BST_UNCHECKED) return wxCHK_UNCHECKED;
>     return wxCHK_UNDEF;
> #else
>     return ((0x001 & SendMessage(GetHwnd(), BM_GETCHECK, 0, 0)) == 0x001);
> #endif
> }
> 
> bool wxCheckBox::Has3State() const
> {
>     return ((m_checkboxStyle & BS_AUTO3STATE) == BS_AUTO3STATE);

*****CVS exited normally with code 1*****


 	  	 
